### ISSUE TITLE
Add iOS 15 extra RAM entitlement to allow 12 GB of RAM when available (Feature Request)

### ISSUE THREAD

**Issue by kgsbowtie** [thebaselab/codeapp#259]
**TL;DR**: iOS 15 has an entitlement for an app to use more RAM. More RAM might be as easy as adding the [`com.apple.developer.kernel.increased-memory-limit`](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_kernel_increased-memory-limit) entitlement set to `true` to the [`Code App.entitlements`](https://github.com/thebaselab/codeapp/blob/9c769cde41a404fd06efb609935320e8296f1db6/Code%20App.entitlements) file.

### Background
The 2021 M1 iPad Pros launched with models that contain 8GB and 16GB of RAM. Previously, apps have been limited in the amount of RAM they can utilize to a maximum of about 5GB, a number completely determined by iOS itself and outside the developer’s control. Recently, with iOS 15, Apple has enabled a way for devices to utilize more RAM if they have it available.

### What needs to be done?
If I correctly understand [Apple’s documentation about the necessary entitlement](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_kernel_increased-memory-limit), all that needs to be done is to add [`com.apple.developer.kernel.increased-memory-limit`](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_kernel_increased-memory-limit) to [`Code App.entitlements`](https://github.com/thebaselab/codeapp/blob/9c769cde41a404fd06efb609935320e8296f1db6/Code%20App.entitlements) (the app’s entitlement list) and set its value to `true`. 

Then use [`os_proc_available_memory`](https://developer.apple.com/documentation/os/3191911-os_proc_available_memory) as necessary (if at all necessary) to check available RAM usage to utilize more RAM if it’s available and ensure that more RAM isn’t being utilized than is available.

### Might be as simple as updating `Code App.entitlements`
So something worth pointing out, the Apple documentation states:
> Add this entitlement to your app to inform the system that some of your app’s core features may perform better by exceeding the default app memory limit on supported devices. If you use this entitlement, make sure your app still behaves correctly if additional memory isn’t available.

To me, it makes it sound like this entitlement can simply be dropped into a working app with no extra work required, implying that iOS will handle things automatically. So if there’s no manual RAM management logic in the app then I’d assume there’s no work to be done here except add the entitlement. Unfortunately, I don’t have a machine to test this on though at the moment. So it may be as simple as updating the entitlements file with no extra work depending on how things are structured.

**Comment by bummoblizard** [comment]
While I too don't have a M1 iPad to test this out, I'll try to add this entitlement in the next release.
